test/useFormState-input.test.js

```diff
@@ -534,23 +534,3 @@ describe('Input blur behavior', () => {
     );
   });
 });
-
-describe('Input props are memoized', () => {
-  it('does not cause re-render of memoized components', () => {
```

Comment:
Thanks for the PR @derrickbeining.

> `describe('Input props are memoized', () => {`
>     `it('does not cause re-render of memoized components', () => {`

It's for this reason the callbacks are being cached (see #39).

Removing the code and _the test_ as a fix to the problem you're running into isn't a preferable solution unfortunately. I expect this use-case stated in the test to _still be supported_. I'm open to any other alternative solutions without compromising memoization. Maybe an implementation similar to `useCallback(callback, deps)` here would helpful.

It's worth mentioning that validate gets passed the form state values as an argument of the form state so you don't have to reach out for the outer scope: https://github.com/wsmd/react-use-form-state#custom-input-validation

Can I ask you to provide an example of the problem you are running into to help better understand your use case? Would passing the entire form state as an argument to validate be helpful?

Reply:
Here is an example of the issue I ran into: https://codesandbox.io/s/3rxryl7yr6

Passing the entire form state to `validate` wouldn't fix the issue because the problem arises when you need to reference values in `validate` other than what's in form state.

I think you should leave caching to users. You can achieve the desired performance optimization simply by using `React.useMemo`, like so:

```typescript
const [formState, inputs] = useFormState({ email: "" })

const emailInput = React.useMemo(() => {
  return inputs.text({
    name: "email",
    validate(val) {
      return ...
    }
  })
}, [])
```

If you wanted `react-use-form-state` to cache things, you'd have accept a dependencies array on each input constructor (`inputs.text` etc), and at that point, it would just be simpler to use `React.useMemo` as shown above.

Reply:
Hmmm you bring up a interesting point about using `useMemo`, but wouldn't memozing the entire  props prevent `value` from being updated when onChange is fired?

```js
const emailInput = React.useMemo(() => {
  return inputs.text({
    name: "email",
    validate(val) {
      return ...
    }
  })
}, [])

emailInput.value // this will never change
```

Here's an exampe: https://codesandbox.io/s/5kxyrmww2l

Reply:
Good point. The current implementation/API appears to not support this performance optimization very well then.

In any case, caching callbacks causes bugs. Removing the cache fixes the bug without a breaking change. To fix the issue while still supporting the caching, you'd probably need to make a breaking change to the API to offer a cache invalidation mechanism.

I think it'd be smart to give up the small performance optimization in order to gain correct functionality.